src/components/core/badge/badge.tsx

```diff
@@ -137,29 +185,60 @@ const BadgeContainer = styled('span', {
 export const Badge = forwardRef<HTMLSpanElement, BadgeProps>(
     (
         {
-            icon_src,
-            icon_class,
+            id,
+            prefix_icon_src,
+            suffix_icon_src,
+            prefix_icon_class,
+            suffix_icon_class,
+            prefixIconOnClickHandler,
+            suffixIconOnClickHandler,
```

Comment:
As mentioned before, best to refactor to accept a `suffix_icon` and `prefix_icon`

Reply:
sorted